main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1553,13 +1553,14 @@ private void uploadPersonalNote() {
                 public void call() {
                     final IConnector con = ConnectorFactory.getConnector(cache);
                     if (con.supportsPersonalNote()) {
-                        con.uploadPersonalNote(cache);
+                        final boolean success = con.uploadPersonalNote(cache);
+
+                        final Message msg = Message.obtain();
+                        final Bundle bundle = new Bundle();
+                        bundle.putString(SimpleCancellableHandler.MESSAGE_TEXT, CgeoApplication.getInstance().getString(success ? R.string.cache_personal_note_upload_done : R.string.cache_personal_note_upload_error));
+                        msg.setData(bundle);
+                        myHandler.sendMessage(msg);
```

Comment:
If this method is called and the connector does not support personal notes, no message would be sent, thus the handler would wait infinitely.

IMHO a message should be sent in any case, either with a success or a failure message (like in this patch) or with a "not supported" message as a dirty way of handling this situation in a case that shouldn't happen (because the button to upload the note should not be visible unless it is possible).